HomeTask 05 (Titanic Dataset)

In [ ]:
import numpy as np
import pandas as pd
titds = pd.read_csv('/content/drive/MyDrive/Colab Notebooks/Titanic-Dataset.csv')
titds

PassengerId  Survived  Pclass  \
0              1         0       3   
1              2         1       1   
2              3         1       3   
3              4         1       1   
4              5         0       3   
..           ...       ...     ...   
886          887         0       2   
887          888         1       1   
888          889         0       3   
889          890         1       1   
890          891         0       3   

                                                  Name     Sex   Age  SibSp  \
0                              Braund, Mr. Owen Harris    male  22.0      1   
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                               Heikkinen, Miss. Laina  female  26.0      0   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                             Allen, Mr. William Henry    male  35.0      0   
..                                                 ...     ...   ...    ...   
886                              Montvila, Rev. Juozas    male  27.0      0   
887                       Graham, Miss. Margaret Edith  female  19.0      0   
888           Johnston, Miss. Catherine Helen "Carrie"  female   NaN      1   
889                              Behr, Mr. Karl Howell    male  26.0      0   
890                                Dooley, Mr. Patrick    male  32.0      0   

     Parch            Ticket     Fare Cabin Embarked  
0        0         A/5 21171   7.2500   NaN        S  
1        0          PC 17599  71.2833   C85        C  
2        0  STON/O2. 3101282   7.9250   NaN        S  
3        0            113803  53.1000  C123        S  
4        0            373450   8.0500   NaN        S  
..     ...               ...      ...   ...      ...  
886      0            211536  13.0000   NaN        S  
887      0            112053  30.0000   B42        S  
888      2        W./C. 6607  23.4500   NaN        S  
889      0            111369  30.0000  C148        C  
890      0            370376   7.7500   NaN        Q  

[891 rows x 12 columns]

In [ ]:
totempty = titds.isnull().sum().sum()
print(totempty)

866


In [17]:
#замена пустых значений на 0
titds.fillna(0, inplace=True)

In [18]:
# общая статистика
titds.describe()

PassengerId    Survived      Pclass         Age       SibSp  \
count   891.000000  891.000000  891.000000  891.000000  891.000000   
mean    446.000000    0.383838    2.308642   23.799293    0.523008   
std     257.353842    0.486592    0.836071   17.596074    1.102743   
min       1.000000    0.000000    1.000000    0.000000    0.000000   
25%     223.500000    0.000000    2.000000    6.000000    0.000000   
50%     446.000000    0.000000    3.000000   24.000000    0.000000   
75%     668.500000    1.000000    3.000000   35.000000    1.000000   
max     891.000000    1.000000    3.000000   80.000000    8.000000   

            Parch        Fare  
count  891.000000  891.000000  
mean     0.381594   32.204208  
std      0.806057   49.693429  
min      0.000000    0.000000  
25%      0.000000    7.910400  
50%      0.000000   14.454200  
75%      0.000000   31.000000  
max      6.000000  512.329200

In [19]:
# уникальные значения
titds.nunique()

PassengerId    891
Survived         2
Pclass           3
Name           891
Sex              2
Age             89
SibSp            7
Parch            7
Ticket         681
Fare           248
Cabin          148
Embarked         4
dtype: int64

**PassengerId	Survived	Pclass	Age	SibSp	Parch	Fare**

**min**	1.000000	0.000000	1.000000	0.000000	0.000000	0.000000	0.000000

**max**	891.000000	1.000000	3.000000	80.000000	8.000000	6.000000	512.329200


In [35]:
titds.corr(method='pearson')

<ipython-input-35-827c5b11c52b>:1: FutureWarning: The default value of numeric_only in DataFrame.corr is deprecated. In a future version, it will default to False. Select only valid columns or specify the value of numeric_only to silence this warning.
  titds.corr(method='pearson')


PassengerId  Survived    Pclass       Age     SibSp     Parch  \
PassengerId     1.000000 -0.005007 -0.035144  0.038125 -0.057527 -0.001652   
Survived       -0.005007  1.000000 -0.338481  0.010539 -0.035322  0.081629   
Pclass         -0.035144 -0.338481  1.000000 -0.361353  0.083081  0.018443   
Age             0.038125  0.010539 -0.361353  1.000000 -0.184664 -0.048786   
SibSp          -0.057527 -0.035322  0.083081 -0.184664  1.000000  0.414838   
Parch          -0.001652  0.081629  0.018443 -0.048786  0.414838  1.000000   
Fare            0.012658  0.257307 -0.549500  0.135516  0.159651  0.216225   

                 Fare  
PassengerId  0.012658  
Survived     0.257307  
Pclass      -0.549500  
Age          0.135516  
SibSp        0.159651  
Parch        0.216225  
Fare         1.000000

Pclass и SibSP отрицательно влияли на выживание при крушении.
(Вероятно, PassengedId при его наличии свидетельствует о более высоком стстусе пассажира и указывает на более высокую их выживаемость)
Более высокая стоимость билета (Fare) положительно влияла нв выживаемость из-за более высокого статуса пассажиров.
